pkg/system_config/types.go

```diff
@@ -0,0 +1,228 @@
+	rc, _ := rsc.registriesMap[registry]
+	if rc == nil {
+		rc = &registryConf{
+			Location: registry,
+		}
+		rsc.registriesMap[registry] = rc
+		rsc.Registries = append(rsc.Registries, rc)
+	}
+	return rc
+}
+
+func (rsc registriesConf) writeToFile() error {
+	return writeTomlFile(RegistriesConfPath, rsc)
+}
+
+func (rsc registriesConf) getRegistryConf(registry string) (*registryConf, bool) {
+	rc, ok := rsc.registriesMap[registry]
+	return rc, ok
+}
+
+type registryConf struct {
+	Location string   `toml:"location"`
+	Prefix   string   `toml:"prefix"`
+	Mirrors  []string `toml:"mirror"`
+	// Setting the blocked, allowed and insecure fields to nil will cause them to be omitted from the output
+	Blocked  *bool `toml:"blocked"`
+	Allowed  *bool `toml:"allowed"`
+	Insecure *bool `toml:"insecure"`
+}
+
+// defaultRegistriesConf returns a default registriesConf object
+func defaultRegistriesConf() registriesConf {
+	return registriesConf{
+		UnqualifiedSearchRegistries: []string{"registry.access.redhat.com", "docker.io"},
+		ShortNameMode:               "",
+	}
+}
+
+const (
+	dockerDaemonTransport = "docker-daemon"
+	dockerTransport       = "docker"
+	atomicTransport       = "atomic"
+)
+
+// {"default":[{"type":"insecureAcceptAnything"}],"transports":{"atomic":{"docker.io":[{"type":"reject"}]},"docker":{"docker.io":[{"type":"reject"}]},"docker-daemon":{"":[{"type":"insecureAcceptAnything"}]}}}
+type policyConf struct {
+	Default    []policyEntry                       `json:"default"`
+	Transports map[string]map[string][]policyEntry `json:"transports"`
+}
+
+func (pc policyConf) resetTransports() {
+	pc.Transports = defaultTransports()
+}
+
+func (pc policyConf) setRejectForRegistry(registry string) {
+	pc.setRejectForRegistryOnTransport(registry, dockerTransport)
+	pc.setRejectForRegistryOnTransport(registry, atomicTransport)
+}
+
+func (pc policyConf) setRejectForRegistryOnTransport(registry, transport string) {
+	pc.Transports[transport][registry] = []policyEntry{
+		rejectPolicyEntry(),
+	}
+}
+
+func (pc policyConf) writeToFile() error {
+	return writeJSONFile(PolicyConfPath, pc)
+}
+
+// defaultPolicyConf returns a default policyConf object
+func defaultPolicyConf() policyConf {
+	return policyConf{
+		Default: []policyEntry{
+			insecureAcceptAnythingPolicyEntry(),
+		},
+		Transports: defaultTransports(),
+	}
+}
+
+func defaultTransports() map[string]map[string][]policyEntry {
+	return map[string]map[string][]policyEntry{
+		dockerDaemonTransport: {
+			"": []policyEntry{
+				insecureAcceptAnythingPolicyEntry(),
+			},
+		},
+		atomicTransport: {},
+		dockerTransport: {},
+	}
+}
+
+func insecureAcceptAnythingPolicyEntry() policyEntry {
+	return policyEntry{
+		Type: "insecureAcceptAnything",
+	}
+}
+
+func rejectPolicyEntry() policyEntry {
+	return policyEntry{
+		Type: "reject",
+	}
+}
+
+type policyEntry struct {
+	Type string `json:"type"`
+}
+
+func writeTomlFile(path string, data interface{}) error {
+	createBaseDir(path)
```

Comment:
any possibility to have these files as mem mapped files too?

Reply:
I initially thought about that too. 
However, the (asynchronous events based) sources are several in this case,  and we still need to take care of concurrency. Also, while the ones we generate for each pod have the potential to become a lot and to strongly impact i/o, the ones deriving from these handlers should be much more stable in the cluster lifetime. Moreover, the custom CAs require a precise FS structure.

Therefore, I decided not to follow that path in this case.. Won't?